src/state/application/hooks.ts

```diff
@@ -320,23 +320,29 @@ export function useETHPrice(version: string = VERSION.CLASSIC): AppState['applic
     if (!isEVM) return
 
     async function checkForEthPrice() {
-      const [newPrice, oneDayBackPrice, pricePercentChange] = await (version === VERSION.ELASTIC
-        ? getPrommEthPrice(chainId, elasticClient, blockClient)
-        : getEthPrice(chainId, classicClient, blockClient))
-
-      dispatch(
-        version === VERSION.ELASTIC
-          ? updatePrommETHPrice({
-              currentPrice: (newPrice ? newPrice : 0).toString(),
-              oneDayBackPrice: (oneDayBackPrice ? oneDayBackPrice : 0).toString(),
-              pricePercentChange,
-            })
-          : updateETHPrice({
-              currentPrice: (newPrice ? newPrice : 0).toString(),
-              oneDayBackPrice: (oneDayBackPrice ? oneDayBackPrice : 0).toString(),
-              pricePercentChange,
-            }),
-      )
+      if (fetchingETHPrice) return
```

Comment:
why dont you use a loading state instead of this?

Reply:
oh, multiple components use this hook simultaneously

Reply:
Yes. Multiple component call this and it just need to load once.